File: engine/model/src/main/scala/tools/packet/movement/MovePlayer.scala
package tools.packet.movement

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class MovePlayer(private var _characterId: Int, private var _movementList: java.util.List[java.lang.Byte]) extends PacketInput {
  def characterId: Int = _characterId

  def movementList: java.util.List[java.lang.Byte] = _movementList

  override def opcode(): SendOpcode = SendOpcode.MOVE_PLAYER
}

File: engine/engine-model/src/main/java/tools/packet/movement/MovePlayer.java
package tools.packet.movement;

import java.util.List;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record MovePlayer(Integer characterId, List<Byte> movementList) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MOVE_PLAYER;
   }
}